core/src/main/scala/org/bitcoins/core/protocol/transaction/Transaction.scala

```diff
@@ -108,40 +118,241 @@ sealed abstract class Transaction extends NetworkElement {
   }
 }
 
-sealed abstract class BaseTransaction extends Transaction {
-  override def bytes = RawBaseTransactionParser.write(this)
-  override def weight = byteSize * 4
+object Transaction extends Factory[Transaction] {
+  override def fromBytes(bytes: ByteVector): Transaction = {
+    Try(WitnessTransaction(bytes)) match {
+      case Success(wtx) => wtx
+      case Failure(_)   => BaseTransaction(bytes)
```

Comment:
This is a current problem, but it'd nice to be fixed here. Currently, when we do 
```
Transaction.fromBytes(EmptyTransaction.bytes)
```
It will return a `BaseTransaction` not an `EmptyTransaction`